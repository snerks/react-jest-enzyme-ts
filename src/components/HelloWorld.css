.HelloWorld {
  border: 2px solid black;
  text-align: center;
  background: #f5f5f5;
  color: #333;
  margin: 20px;
  padding: 20px;
}