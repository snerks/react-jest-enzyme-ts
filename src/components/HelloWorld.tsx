import * as React from 'react';
import './HelloWorld.css';

export interface HelloWorldProps {
    name: string;
    removeGreeting: Function;
}

export interface HelloWorldState {
    greeting: string;
}

class HelloWorld extends React.Component<HelloWorldProps, HelloWorldState> {

    constructor(props: HelloWorldProps) {
        super();

        this.state = { greeting: 'Hello' };
    }

    frenchify() {
        this.setState({ greeting: 'Bonjour' });
    }

    removeGreeting() {
        this.props.removeGreeting(this.props.name);
    }

    render() {
        return (
            <div className="HelloWorld">
                {this.state.greeting} {this.props.name}!
                <br />
                <button className="frenchify" onClick={() => this.frenchify()}>Frenchify!</button>
                <br />
                <button className="remove" onClick={() => this.removeGreeting()}>Remove Me!</button>
            </div>
        );
    }
}

export default HelloWorld;
