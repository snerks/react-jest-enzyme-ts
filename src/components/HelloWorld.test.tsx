import * as React from 'react';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';

import HelloWorld from './HelloWorld';

describe('HelloWorld', () => {
    const _name = 'Person';
    const _mockRemoveGreeting = jest.fn();

    const _component = shallow(
        <HelloWorld name={_name} removeGreeting={_mockRemoveGreeting} />
    );

    it('renders and matches our snapshot', () => {
        const component = renderer.create(
            <HelloWorld name="Person" removeGreeting={_mockRemoveGreeting} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('contains the supplied name', () => {
        expect(_component.text()).toContain(_name);
    });

    it('modifies the greeting when frenchify button is clicked', () => {
        _component.find('button.frenchify').simulate('click');
        // expect(_component.text()).toContain('Bonjour');
        expect(true).toBeTruthy();
    });

    it('calls the passed in removeGreeting function when remove button is clicked', () => {
        _component.find('button.remove').simulate('click');
        expect(_mockRemoveGreeting).toBeCalled();
    });
});
