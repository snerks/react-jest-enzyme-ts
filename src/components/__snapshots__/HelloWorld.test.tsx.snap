exports[`HelloWorld renders and matches our snapshot 1`] = `
<div
  className="HelloWorld">
  Hello
   
  Person
  !
  <br />
  <button
    className="frenchify"
    onClick={[Function]}>
    Frenchify!
  </button>
  <br />
  <button
    className="remove"
    onClick={[Function]}>
    Remove Me!
  </button>
</div>
`;
